analyzer/diagnostics.go

```diff
@@ -0,0 +1,26 @@
+package analyzer
+
+import (
+	"golang.org/x/tools/go/analysis"
+)
+
+func newAnalysisDiagnostic(
+	checker string,
```

Comment:
Can we introduce constants for checker? Looks like we have a limited number of possible checkers.